### PR TITLE
e2e: test Update with GH reference and context dir validation

### PR DESCRIPTION
**Pull request by subhashkhileri**
## Fixes 

https://issues.redhat.com/browse/HAC-3184
https://issues.redhat.com/browse/HAC-3779

## Description

1. e2e:Update resolved issue tests case
2. Update e2e test to repo with 2 components with GH reference and Context Dir validation

## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Test

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: integration-tests/tests/create-application-from-public-git-source.spec.ts [hunk 1]
```diff
@@ -41,12 +44,9 @@ describe('Create Component from Public Git Source', { tags: ['@PR-check', '@publ
       Common.waitForLoad();
```
Comment (Katka92) [inline integration-tests/tests/create-application-from-public-git-source.spec.ts:44]: IDK, is this wait needed there? We are still on the same page.

File: integration-tests/tests/create-application-from-public-git-source.spec.ts [hunk 2]
```diff
@@ -5,13 +5,16 @@ import { ApplicationDetailPage } from '../support/pages/ApplicationDetailPage';
...
 
 describe('Create Component from Public Git Source', { tags: ['@PR-check', '@publicRepo'] }, () => {
   const addComponent = new AddComponentPage();
   const componentPage = new ComponentPage();
   const applicationDetailPage = new ApplicationDetailPage();
   const applicationName = Common.generateAppName();
-  const publicRepo = 'https://github.com/dheerajodha/devfile-sample-code-with-quarkus';
+  const publicRepo = 'https://github.com/hac-test/nodejs-ex';
+  const gitReference = 'v1.0.1';
+  const contextDir = './';
```
Comment (Katka92) [inline integration-tests/tests/create-application-from-public-git-source.spec.ts:17]: Can we have some real contextDir? E.g. in some other repo? I was thinking about hac-dev repo and integration-tests context dir, for example.

File: integration-tests/tests/create-application-from-public-git-source.spec.ts [hunk 3]
```diff
@@ -86,15 +81,17 @@ describe('Create Component from Public Git Source', { tags: ['@PR-check', '@publ
...
-    // it('Change Resources Value', () => {
-    //   applicationDetailPage.openComponentSettings(componentPage.componentName);
-    //   componentPage.setRam(2, MemoryUnit.gigabyte);
-    //   componentPage.setCpuByButton(cpuCount, cpuUnit);
-    //   componentPage.saveChanges();
-    //   Applications.goToComponentsTab();
-    //   applicationDetailPage.expandDetails(componentPage.componentName);
-    //   applicationDetailPage.checkCpuAndMemory(cpuCount, CPUUnit.millicore, 2, MemoryUnit.gigabyte);
-    // });
+    it('Change Resources Value', () => {
```
Comment (Katka92) [inline integration-tests/tests/create-application-from-public-git-source.spec.ts:84]: What about having two last tests in a different suite then `Creating a Quarkus Component`, as the component is already created?

**Comment by Katka92** [comment]
Could you please also delete [this](https://github.com/openshift/hac-dev/pull/576/files#diff-020077030fb4559e7df09c79bb427382bb75fd136bb7b389c35c90f88a0d6bc6R31) line as part of this PR? :slightly_smiling_face: thanks

File: integration-tests/tests/create-application-from-public-git-source.spec.ts (hunk 1 above)
Comment (subhashkhileri) [reply integration-tests/tests/create-application-from-public-git-source.spec.ts:44]: removed

File: integration-tests/tests/create-application-from-public-git-source.spec.ts (hunk 3 above)
Comment (subhashkhileri) [reply integration-tests/tests/create-application-from-public-git-source.spec.ts:84]: I didn't understood, though there is `Creating a Quarkus Component` which I will update to `Creating a Nodejs Component` as I have update the repo from Quarkus to nodejs

File: integration-tests/tests/create-application-from-public-git-source.spec.ts (hunk 2 above)
Comment (subhashkhileri) [reply integration-tests/tests/create-application-from-public-git-source.spec.ts:17]: Okay, you want be to create a new dir and have nodejs-ex in it and pass dir name as contextDir?

Comment (Katka92) [reply integration-tests/tests/create-application-from-public-git-source.spec.ts:17]: IDK, or use some repo where we can actually test this contextDir. But if you think it is out of the scope of this PR, we can just create an issue for that :slightly_smiling_face:

File: integration-tests/tests/create-application-from-public-git-source.spec.ts (hunk 3 above)
Comment (Katka92) [reply integration-tests/tests/create-application-from-public-git-source.spec.ts:84]: I mean the component is created on line 73. So, the changes of resources done in the following two tests are changing the resources of an existing app. So I propose doing something like that:

line 78:
```
describe('Change resources for the existing app', () => {
   it('Check Resources Value', () => {
       ...
   });

   it('Change Resources Value', () => {
      ...
   });
});
```

**Pull request merged**